core/incremental/cursor.rs

```diff
@@ -174,10 +174,26 @@ impl MaterializedViewCursor {
                     let btree_result =
                         return_if_io!(self.btree_cursor.seek(SeekKey::TableRowId(target), op));
 
-                    let changes = if btree_result == SeekResult::Found {
-                        return_if_io!(self.read_btree_delta_entry())
-                    } else {
-                        Vec::new()
+                    let changes = match btree_result {
+                        SeekResult::Found => return_if_io!(self.read_btree_delta_entry()),
+                        SeekResult::TryAdvance => {
+                            // Cursor is positioned at the leaf but current entry doesn't match.
+                            // Advance in the appropriate direction to find the next matching entry.
+                            let advanced = match op {
+                                SeekOp::GT | SeekOp::GE { .. } => {
+                                    return_if_io!(self.btree_cursor.next())
```

Comment:
Actually this is still a bit inaccurate because if this returns IO here, we will end up seeking again on line 175....